### PR TITLE
Categories for events as well as ability to filter with these categories

### PR DESCRIPTION
**Pull request by bradleyyma**
Added a few categories for our events. I have the tuples coded locally since I'm not sure how to set up a global set of CATEG(ories). I also added checkboxes for the user to filter their search based on their choices of categories. I did some basic manual tests and it looks like it is functionally properly (hopefully). 

Similar to the issue with the sorting form through crispy, I didn't format the checkboxes through crispy either since I can't seem to get them inline.

Database is updated with more events and existing events have their categories updated to follow the new selection system.

### CODE REVIEW

File: events/models.py [hunk 1]
```diff
@@ -6,7 +6,16 @@
...
+    CATEG=[
+        ('Basketball', 'Basketball'),
+        ('Cycling', 'Cycling'),
+        ('Hiking', 'Hiking'),
+        ('Running', 'Running'),
+        ('Soccer', 'Soccer'),
+        ('Tennis', 'Tennis'),
+        ('Other', 'Other'),
+    ]
+    category = models.CharField(max_length=100, choices=CATEG) # TODO: Implement choices set for category
```
Comment (royceschultz) [inline events/models.py]: remove TODO

File: events/views.py [hunk 2]
```diff
@@ -21,6 +21,25 @@ def home(request):
...
+            'Soccer',
+            'Tennis',
+            'Other',
+            ]
+        filtered_events = Event.objects.none()
+        #filter for each category chosen
+        for i in CATEG:
+            if i in category:
+                filtered_events = filtered_events|events.filter(category = i)
+        events = filtered_events
```
Comment (royceschultz) [inline events/views.py]: Use the [dunder in](https://stackoverflow.com/questions/9304908/how-can-i-filter-a-django-query-with-a-list-of-values) method.
More about [dunder methods](https://www.geeksforgeeks.org/dunder-magic-methods-python/). It's REALLY useful for defining classes.

File: events/templates/home.html [hunk 3]
```diff
@@ -5,7 +5,10 @@ <h1 class="">Events</h1>
     <form class="d-inline" action="{% url 'home'%}" method="get">
       <!-- {% csrf_token %} -->
       <fieldset>
-        {{ form|crispy }}
+        {{ form.search|as_crispy_field }}
+        {{ form.show_old|as_crispy_field }}
+        {% include "widgets/filterOptions.html" %}
```
[the remark this answers is not in the document]
Comment (bradleyyma) [reply events/templates/home.html]: Can you elaborate a bit more, or send a link. I thought widget classes could only be used inside the form.py file.

[the remark this answers is not in the document]
Comment (royceschultz) [reply events/templates/home.html]: They're called in the form file, but we can define custom widgets in any file that we can import. The datetime picker is defined in a file called 'widgets.py' although the name is arbitrary.
This implementation is kinda like manually selecting the widget via the html file. It's functional and thats all that matters, but if we want a less verbose and reusable component, the rendering should be done in the definition, not the html implementation.

**Pull request merged**